types/options.go

```diff
@@ -310,10 +311,27 @@ func getRootlessStorageOpts(rootlessUID int, systemOpts StoreOptions) (StoreOpti
 	}
 	if opts.GraphDriverName == "" {
 		if len(systemOpts.GraphDriverPriority) == 0 {
-			if canUseRootlessOverlay(opts.GraphRoot, opts.RunRoot) {
-				opts.GraphDriverName = overlayDriver
-			} else {
-				opts.GraphDriverName = "vfs"
+			driversMap := drivers.ScanPriorDrivers(opts.GraphRoot)
+
+			// use the supplied priority list unless it is empty
+			prioList := systemOpts.GraphDriverPriority
+			if len(prioList) == 0 {
```

Comment:
how can this be longer than 0 if we just checked for it at line 313?

Reply:
This check is not needed. I missed it while copying it from drivers.